execute/selector.go

```diff
@@ -139,23 +141,23 @@ func (t *indexSelectorTransformation) Process(id DatasetID, tbl flux.Table) erro
 		return fmt.Errorf("unsupported selector type %v", valueCol.Type)
 	}
 
-	return tbl.Do(func(cr flux.ColReader) error {
+	return tbl.DoArrow(func(cr flux.ArrowColReader) error {
 		switch valueCol.Type {
 		case flux.TBool:
 			selected := s.(DoBoolIndexSelector).DoBool(cr.Bools(valueIdx))
-			return t.appendSelected(selected, builder, cr)
+			return t.appendSelected(selected, builder, arrow.ColReader(cr))
```

Comment:
`appendSelected` should be modified to use the arrow column reader. I don't think there is anybody else calling it  and `arrow.ColReader` should be used sparingly because it is being deleted when we're done with this transition.

Reply:
@jsternberg I should have written a comment or TODO here. Modifying `appendSelected` would indirectly require a modification to `execute.AppendRecord` and `execute.ValueForRow` which multiple functions use. So I decided it was probably best left untouched for right now.

Reply:
Whenever you run into those kinds of functions, can you create an arrow version of them and say something like `AppendArrowRecord`?

That way we can do the same thing we will do with `DoArrow`. The main reason is because with this method, we have to manually go in and find the usages of `ColReader` and change them all. If we duplicate the `execute` helper methods, we can use GoLand to determine that nobody  uses the original method and then use refactor to change everyone over at the same time in a more mechanical process.